pkg/ansible/generate.go

```diff
@@ -0,0 +1,257 @@
+# openshift cluster.
+
+# must specify a base_ami when building an AMI
+openshift_aws_base_ami: ami-ae7bfdb8 # ami-12345678
+
+# this is the ami we built
+openshift_aws_ami: ami-9a0d95e0 # ami-2f53ce55
+
+# when creating an encrypted AMI please specify use_encryption
+#openshift_aws_ami_encrypt: False
+
+# -- #
+# S3 #
+# -- #
+
+# Create an s3 bucket.
+openshift_aws_create_s3: false
+
+# --- #
+# ELB #
+# --- #
+
+# openshift_aws_elb_name will be the base-name of the ELBs.
+# TODO: looks like this is supposed to just be basename variant
+#openshift_aws_elb_name: dgoodwin-dev
+
+# custom certificates are required for the ELB
+openshift_aws_iam_cert_path: /home/abutcher/rhat/certs/wildcard-flibberty-jibbet.com.crt  # '/path/to/wildcard.<clusterid>.example.com.crt'
+openshift_aws_iam_cert_key_path: /home/abutcher/rhat/certs/wildcard-flibberty-jibbet.com.key  # '/path/to/wildcard.<clusterid>.example.com.key'
+openshift_aws_iam_cert_chain_path: /home/abutcher/rhat/certs/ca.crt  # '/path/to/cert.ca.crt'
+
+openshift_aws_create_iam_role: True
+
+{{ range $key, $value := . }}
+   {{ $key }}: {{ $value }}
+{{ end }}
+`
+
+// TODO: these {{}} don't play nice in a go template, complete this when we get to processing
+// node groups dynamically
+/*
+# tags
+openshift_aws_master_group_tags:
+  Name: "{{ openshift_aws_clusterid }}-master"
+  host-type: master
+  sub-host-type: default
+
+openshift_aws_node_group_compute_tags:
+  Name: "{{ openshift_aws_clusterid }}-compute"
+  host-type: node
+  sub-host-type: compute
+
+openshift_aws_node_group_infra_tags:
+  Name: "{{ openshift_aws_clusterid }}-infra"
+  host-type: node
+  sub-host-type: infra
+*/
+
+const (
+	cloudProviderKindVar = "openshift_cloudprovider_kind"
+	awsClusterIDVar      = "openshift_aws_clusterid"
+	elbBasenameVar       = "openshift_aws_elb_basename"
+	awsVPCNameVar        = "openshift_aws_vpc_name"
+	awsRegionVar         = "openshift_aws_region"
+	awsBaseAMINameVar    = "openshift_aws_base_ami_name"
+)
+
+type VarsGenerator struct {
+	Vars map[string]string
+}
+
+func NewVarsGenerator(cluster *coapi.Cluster) *VarsGenerator {
+
+	// Any vars that change from cluster to cluster need to be defined here:
+	inv := &VarsGenerator{
+		Vars: map[string]string{
+			cloudProviderKindVar: "aws",
+			awsRegionVar:         "us-east-1",
+			awsClusterIDVar:      cluster.Name,
+			elbBasenameVar:       cluster.Name,
+			awsVPCNameVar:        cluster.Name,
+			awsBaseAMINameVar:    fmt.Sprintf("%s-ami-base", cluster.Name),
+		},
+	}
+	return inv
+}
+
+// GenerateVars returns a string suitable for use as a vars file when calling openshift-ansible pods.
+func (ig *VarsGenerator) GenerateVars() (string, error) {
+	/*
+		var buffer bytes.Buffer
+
+		for k, v := range ig.Vars {
+			buffer.WriteString(fmt.Sprintf("%s=%s", k, v))
+		}
+
+		return fmt.Sprintf(invTemplate, buffer.String())
+	*/
+	t := template.Must(template.New("varsfilecontents").Parse(varsTemplate))
```

Comment:
As `GenerateVars` returns an error to pass along problems, it would be more appropriate to return any error from `Parse` rather than panicking here.

Reply:
Agreed, adjusting.